test/integration_test/clusterdomain_test.go

```diff
@@ -0,0 +1,179 @@
+	t *testing.T,
+	name string,
+	domain string,
+	active bool,
+) {
+	setRemoteConfig("")
+
+	_, err := k8s.Instance().CreateClusterDomainUpdate(&v1alpha1.ClusterDomainUpdate{
+		ObjectMeta: meta.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1alpha1.ClusterDomainUpdateSpec{
+			ClusterDomain: domain,
+			Active:        active,
+		},
+	})
+	require.NoError(t, err, "Unexpected error on cluster domain update: %v", err)
+}
+
+func failoverAndFailbackClusterDomainTest(t *testing.T) {
+	// validate the cluster domains status
+	err := k8s.Instance().ValidateClusterDomainsStatus(cdsName, domainMap, defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "validation of cluster domain status for %v failed", cdsName)
+
+	// Migrate the resources
+	ctxs, preMigrationCtx := triggerMigration(
+		t,
+		"mysql-clusterdomain-migration",
+		"mysql-1-pvc",
+		nil,
+		[]string{"mysql-clusterdomain-migration"},
+		true,
+		true,
+	)
+
+	// validate the following
+	// - migration is successful
+	// - app starts on cluster 1
+	validateAndDestroyMigration(t, ctxs, preMigrationCtx, true, false, false, true)
+
+	testClusterDomainsFailover(t, preMigrationCtx, ctxs)
+
+	testClusterDomainsFailback(t, preMigrationCtx, ctxs)
+
+}
+
+func testClusterDomainsFailover(
+	t *testing.T,
+	preMigrationCtx *scheduler.Context,
+	ctxs []*scheduler.Context,
+) {
+	// Failover the application
+
+	// Deactivate cluster domain
+	triggerClusterDomainUpdate(
+		t,
+		deactivateName,
+		domainList[0],
+		false,
+	)
+
+	// build the expected domain map
+	domainMap[domainList[0]] = false
+	// validate the cluster domains status
+	err := k8s.Instance().ValidateClusterDomainsStatus(cdsName, domainMap, defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "validation of cluster domain status for %v failed", cdsName)
+
+	// Reduce the replicas on cluster 1
+	scaleFactor, err := schedulerDriver.GetScaleFactorMap(ctxs[0])
+	require.NoError(t, err, "Unexpected error on GetScaleFactorMap")
+	for k, _ := range scaleFactor {
+		scaleFactor[k] = 0
+	}
+	schedulerDriver.ScaleApplication(ctxs[0], scaleFactor)
+
+	// start the app on cluster 2
+	err = setRemoteConfig(remoteFilePath)
+	require.NoError(t, err, "Error setting remote config")
+
+	// Increase the replicas on cluster 1
+	scaleFactor, err = schedulerDriver.GetScaleFactorMap(preMigrationCtx)
+	require.NoError(t, err, "Unexpected error on GetScaleFactorMap")
+	for k, _ := range scaleFactor {
+		scaleFactor[k] = 1
+	}
+	schedulerDriver.ScaleApplication(preMigrationCtx, scaleFactor)
+
+	err = schedulerDriver.WaitForRunning(preMigrationCtx, defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for pod to get to running state on remote cluster after migration")
+}
+
+func testClusterDomainsFailback(
+	t *testing.T,
+	preMigrationCtx *scheduler.Context,
+	ctxs []*scheduler.Context,
+) {
+	// Failback the application
+
+	// destroy the app on cluster 2
+	err := schedulerDriver.Destroy(preMigrationCtx, nil)
```

Comment:
destroyAndWait() does these both steps, you can use that if you want.

Reply:
It also deletes the PVC.